cmd/fabric8-jenkins-proxy/main_cors_test.go

```diff
@@ -6,46 +6,47 @@ import (
 	"net/http/httptest"
 	"testing"
 
-	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/api"
-	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/proxy"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/testutils/mock"
+
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/clients"
 	"github.com/julienschmidt/httprouter"
 	uuid "github.com/satori/go.uuid"
 	"github.com/stretchr/testify/assert"
 )
 
-type MockProxyAPI interface {
-	Info(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+type MockJenkinsAPI interface {
+	Start(w http.ResponseWriter, r *http.Request, _ httprouter.Params)
 }
 
-type MockProxyAPIImpl struct{}
+type MockJenkinsAPIImpl struct{}
 
-func (m MockProxyAPIImpl) Info(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
-	resp := api.Response{}
+func (api *MockJenkinsAPIImpl) Start(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	resp := clients.StatusResponse{}
 	json.NewEncoder(w).Encode(resp)
 }
 
 func TestAPIServerCORSHeaders(t *testing.T) {
-	apiServer := newAPIServer(MockProxyAPIImpl{})
-	reader, _ := http.NewRequest("GET", "/", nil)
-	randomOrigin := uuid.NewV4().String()
-	reader.Header.Set("origin", randomOrigin) // allowing everything for now.
+	config := mock.NewConfig()
+	apiServer := newJenkinsAPIServer(&MockJenkinsAPIImpl{}, &config)
 
+	reader, _ := http.NewRequest("POST", "/jenkins/start", nil)
```

Comment:
isn't the endpoint `/api/jenkins/start` ?

Reply:
You are right. This has to change. The test will pass because we are not testing the API. We are merely testing if this server allows external requests. So, the endpoint here will not matter.